modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp

```diff
@@ -201,9 +201,9 @@ boolean limitToOneSubmissionPerUser = DDMFormInstanceSubmissionLimitStatusUtil.i
 							<clay:container-fluid>
 								<div class="locale-actions">
 									<liferay-ui:language
-										formAction="<%= currentURL %>"
 										languageId="<%= languageId %>"
 										languageIds="<%= ddmFormDisplayContext.getAvailableLanguageIds() %>"
+										useNamespace="<%= false %>"
```

Comment:
My idea was to make the Language Selector shown in the shared form behave like the ones added to a layout and use the formAction that changes the language at portal level (`com.liferay.portal.action.UpdateLanguageAction`) using urls like `"${portalURL}/c/portal/update_language?languageId=${locale}"`. Doing this, it uses the already existing logic to add the language attribute to the html tag, if the url contains a locale it is also changed and it also translates the submit button which was not being translated now. It also updates the user locale so this language selection is persisted. These last two things are what broke some Poshi tests that I'm working on fixing (I guess I should fix them, shouldn't I?)

For this I had to remove the formAction so that the taglib [uses its default one that is /portal/update_language.](https://github.com/liferay/liferay-portal/blob/master/util-taglib/src/com/liferay/taglib/ui/LanguageTag.java#L160-L172)

I added `useNamespace=false` because it defaults to **true**. If it is true, when calculating each language url the [portlet namespace is added to the parameters of the url](https://github.com/liferay/liferay-portal/blob/master/util-taglib/src/com/liferay/taglib/ui/LanguageTag.java#L231-L234). Resulting in non-working urls like: `/c/portal/update_language?_com_liferay_dynamic_data_mapping_form_web_portlet_DDMFormPortlet_languageId=es_ES`
If set to **false**, portlet namespace is not added to these parameters so the language url is like: `/c/portal/update_language?languageId=en_US`

Reply:
Hi @JavierMoral 
Thanks for explaining.

> It also translates the submit button which was not being translated now.

If I remember correctly, it was decided for https://issues.liferay.com/browse/LPS-151677 that the submit button should be translated based on the **User** language preferences. So I believe that this behavior should not be changed. 

Do you need to change this in order to fix the bug? If so, I suggest opening a PTR to clarify the expected behavior.

Reply:
Hi @carolmariaabb , that's not the behavior that I'm changing. When using the shared form url, the submit button is still being translated to the user language.
What I changed is that, when selecting a language using the language selector shown in the shared form url, the user language is also changed accordingly. This way, the language attribute of the html tag, the submit button and the form fields are consistent.
Having 'en-US' in the language attribute of the html tag (showing the submit button in english) and then rendering the form fields in portuguese imo will cause accessibility problems for people using assistive technologies. Quoting WCAG [Understanding Success Criterion 3.1.1: Language of Page](https://www.w3.org/WAI/WCAG21/Understanding/language-of-page.html):

> Both assistive technologies and conventional user agents can render text more accurately when the language of the Web page is identified. Screen readers can load the correct pronunciation rules. Visual browsers can display characters and scripts correctly. Media players can show captions correctly. As a result, users with disabilities will be better able to understand the content.

Reply:
Hi @JavierMoral,
I understand your point and I agree that the new behavior introduced by you makes more sense.

However, as already mentioned, the decision made by the product team was to not translate the `submit` `previous` and `next` buttons, as you can see in the video below, when changing the language. 

[Gravação de tela de 03-10-2022 17:07:35.webm](https://user-images.githubusercontent.com/34186632/193679057-d03b88fa-498b-4d62-a55b-c12bcd2b4614.webm)

The expected result is to use the form default language and that's why some Poshi tests are failing in your PR.

That said, I think it's important to create a PTR explaining your changes

> What I changed is that, when selecting a language using the language selector shown in the shared form url, the user language is also changed accordingly.

so we can align the decision with the product team.

What do you think?

Reply:
Hi @carolmariaabb , I understand. I opened https://issues.liferay.com/browse/PTR-3388 to discuss the expected behavior.
Thanks!

Reply:
Hi @JavierMoral,
Feel free to fix the Poshi tests and send another PR.

Thanks for working on this.